config/config_test.go

```diff
@@ -71,58 +72,233 @@ func TestCreateConfig(t *testing.T) {
 	})
 
 	t.Run("failure by invalid current", func(t *testing.T) {
-		cnfPath := filepath.Join(testDir, "failureCurrentConfig")
-		_, err := New(cnfPath)
+		config := Config{
+			Entries: map[string]*Entry{
+				"default": {
+					APIURL:   "api-url",
+					StoreURL: "store-api-url",
+					Token:    "dummy_token",
+					Launcher: Launcher{
+						Version: "latest",
+						Image:   "screwdrivercd/launcher",
+					},
+				},
+			},
+			Current: "doesnotexist",
+		}
+
+		_, err := config.Get(config.Current)
 
 		assert.Equal(t, "config `doesnotexist` does not exist", err.Error())
 	})
+}
 
-	t.Run("failure by invalid yaml", func(t *testing.T) {
-		cnfPath := filepath.Join(testDir, "failureConfig")
-		_, err := New(cnfPath)
+func TestConfigAdd(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		config := Config{
+			Current: "default",
+			Entries: map[string]*Entry{
+				"default": {
+					APIURL:   "api-url",
+					StoreURL: "store-api-url",
+					Token:    "dummy_token",
+					Launcher: Launcher{
+						Version: "latest",
+						Image:   "screwdrivercd/launcher",
+					},
+				},
+			},
+		}
 
-		assert.Equal(t, 0, strings.Index(err.Error(), "failed to parse config file: "), fmt.Sprintf("expected error is `failed to parse config file: ...`, actual: `%v`", err))
+		err := config.Add("test")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		expected := Config{
+			Entries: map[string]*Entry{
+				"default": {
+					APIURL:   "api-url",
+					StoreURL: "store-api-url",
+					Token:    "dummy_token",
+					Launcher: Launcher{
+						Version: "latest",
+						Image:   "screwdrivercd/launcher",
+					},
+				},
+				"test": {
+					APIURL:   "",
+					StoreURL: "",
+					Token:    "",
+					Launcher: Launcher{
+						Version: "stable",
+						Image:   "screwdrivercd/launcher",
+					},
+				},
+			},
+			Current: "default",
+		}
+
+		assert.Equal(t, expected, config)
+	})
+
+	t.Run("failure by the name that exists", func(t *testing.T) {
+		config := Config{
+			Current: "default",
+			Entries: map[string]*Entry{
+				"default": {
+					APIURL:   "api-url",
+					StoreURL: "store-api-url",
+					Token:    "dummy_token",
+					Launcher: Launcher{
+						Version: "latest",
+						Image:   "screwdrivercd/launcher",
+					},
+				},
+			},
+		}
+
+		err := config.Add("default")
+		assert.Equal(t, "config `default` already exists", err.Error())
 	})
 }
 
-func TestSetConfig(t *testing.T) {
-	rand.Seed(time.Now().UnixNano())
-	cnfPath := filepath.Join(testDir, ".sdlocal", fmt.Sprintf("%vconfig", rand.Int()))
-	defer os.RemoveAll(filepath.Join(testDir, ".sdlocal"))
+func TestConfigSave(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		rand.Seed(time.Now().UnixNano())
+		cnfPath := filepath.Join(testDir, ".sdlocal", fmt.Sprintf("%vconfig", rand.Int()))
+		err := os.MkdirAll(filepath.Join(testDir, ".sdlocal"), 0777)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer os.RemoveAll(filepath.Join(testDir, ".sdlocal"))
+
+		file, err := os.Create(cnfPath)
```

Comment:
Don't we need to add the test case that Save() create the config file, if there is no config file?

Reply:
I think when the Config object was created, it is guaranteed that the config file exists.
So, we don't use `os.O_CREATE` in `os.OpenFile()` function of `Config.Save()` function.

Reply:
I see config.New guranteed to create config file.
https://github.com/screwdriver-cd/sd-local/blob/master/config/config.go#L26

Thanks!